### DISCUSSION TITLE
Add checkbox vs radiobox behavior option to toggle group component

### DISCUSSION THREAD

**Discussion by btroop** [chakra-ui/zag#1609] [Ideas]
Currently the toggle group can be toggled between single and multi-select. However, the single select does not offer true radio button functionality, as it is possible to "uncheck" one of the options. Would be great to have the ability to flag the toggle group to behave more like a radio group, where once an option is selected the toggle group must always have a toggle selected (until reset).

**Comment by segunadebayo** [comment]
Interesting use case. I've got two questions:

- Did you consider using the Radio or Checkbox machine instead?
- How about using the `api` of the toggle group to design this behavior to your taste?

**Reply by btroop** [reply]
I'm using park ui built on top of ark ui...so i'm not very familiar with the underlying tech/logic and how to implement changes to it. All i know is that they are built on top of zag, so i figured the change would have to happen at the zag level

**Comment by segunadebayo** [comment]
Closing — the toggle-group machine now has `multiple` and `deselectable` props that cover checkbox vs radio behavior.
